### PR TITLE
Show password as Text toggle input

### LINKED ISSUE #515: Show password as Text toggle Input
**Describe the solution you'd like**
There should be a button that allows the user to view a password Text Input as a regular text input. This is useful for users to verify that their value is correct

**Additional context**
It looks like there may be an eye icon that we can use. We will also need the eye icon with a cross through it, indicating the password text is not visible.

### PR DESCRIPTION
**Pull request by msankaran0712**
Inspired by [Christian's library](https://github.com/coliff/bootstrap-show-password-toggle)

Fixes #515 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (including CHANGELOG updates)
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by coliff** [comment]
https://icy-ground-0374a1310-917.centralus.1.azurestaticapps.net/?path=/story/user-inputs-text-input--default&args=type:password

This is awesome. Works great. I think it's fine to be on by default, but I wonder if any developers would ever want to disable the password toggle option from ever appearing? If so, should there be an option - or maybe it's not needed?

**Review by coliff — DISMISSED** [review]

**Comment by msankaran0712** [comment]
> https://icy-ground-0374a1310-917.centralus.1.azurestaticapps.net/?path=/story/user-inputs-text-input--default&args=type:password
> 
> This is awesome. Works great. I think it's fine to be on by default, but I wonder if any developers would ever want to disable the password toggle option from ever appearing? If so, should there be an option - or maybe it's not needed?

Added a prop `includePasswordTextToggle` set true by default. Users can use this prop to hide the toggle if they want.

**Review by coliff — DISMISSED** [review]

File: stencil-workspace/src/components/modus-text-input/modus-text-input.tsx [hunk 1]
```diff
@@ -139,13 +166,53 @@ export class ModusTextInput {
...
-            </span>
-          ) : (
-            <span class="icons"></span>
+          {showPasswordToggle && (
+            <div
+              class="icons toggle-password"
+              aria-label="Show password as plain text. Warning: this will display your password on the screen."
+              ref={(el) => (this.buttonTogglePassword = el as HTMLDivElement)}
+              onClick={() => this.handleTogglePassword()}>
+              <svg
```
Comment (ryan-henness-trimble) [inline stencil-workspace/src/components/modus-text-input/modus-text-input.tsx:175]: We should add these svgs to their own icon files

Comment (msankaran0712) [reply stencil-workspace/src/components/modus-text-input/modus-text-input.tsx]: @ryan-henness-trimble Thanks for spotting that one. Added those icons to their own files now. 
cc @coliff

**Comment by coliff** [comment]
sorry for late feedback on this, but I realised that I think the icons are displayed reversed - they are showing the state of the icons and not the effect they will have if clicked. I made this mistake in the first version of my plugin but changed it after a user noticed it and made a PR.
REF: `https://github.com/coliff/bootstrap-show-password-toggle/pull/83`
Making the change will match the UI that Microsoft IE 11 and Edge use for their built-in 'show password toggle'

**Comment by msankaran0712** [comment]
> sorry for late feedback on this, but I realised that I think the icons are displayed reversed - they are showing the **state** of the input and not the **effect** they will have if clicked. I made this mistake in the first version of my plugin but changed it after a user noticed it and made a PR. REF: `https://github.com/coliff/bootstrap-show-password-toggle/pull/83`
> 
> Making the change will match the UI that Microsoft IE 11 and Edge use for their built-in 'show password toggle'

Reversed the states now, please check.

**Review by coliff — APPROVED** [review]

**Review by ryan-henness-trimble — APPROVED** [review]

**Review by enowak1031 — DISMISSED** [review]

**Review by coliff — APPROVED** [review]

**Pull request merged**